aten/src/ATen/TensorIterator.cpp

```diff
@@ -1281,14 +1294,51 @@ void TensorIteratorBase::build(TensorIteratorConfig& config) {
   view_offsets_ = DimVector(ndim_offsets, 0);
 }
 
+void TensorIteratorBase::set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides, TensorOptions options, DimnameList names) {
```

Comment:
I think it's worth a comment above each of these `set_output` overrides with the brief description of the calling contexts from the PR description.